### DISCUSSION TITLE
Pictures of setups on your boats

### DISCUSSION THREAD

**Discussion by mgrouch** [bareboat-necessities/lysmarine_gen#417] [General]
Please post pictures of you actual setup on a boat in this thread for others to give them ideas how hardware can be arranged on their boat.

**Comment by mgrouch** [comment]
Some picture here:
https://www.thehulltruth.com/marine-electronics-forum/1320681-smart-boat-projects.html

**Comment by atchik6056** [comment]
ready to be installed on the boat
- lysmarine-bbn-full-bookworm_2024-04-20-r1252
- RPI4B 8GB
- Geekworm X735 V3 - power supply/regulator with momentary switch
- Waveshare USB 3.2 HAT
- Raspberry Pi 4 Model B PWM Transparent Acrylic Case, heavily modified with dremel multitool - aliexpress
- WT070B24,Resolution:1024x600,Screen Size: 7" - alibaba
- GeeekPi Raspberry Pi X862 m.2 NGFF 2280 SATA SSD Expansion Board with USB 3.1 -aliexpress
- BR M2 NGFF SATA3 SSD 256GB M.2 N SSD SATA 2280 HDD Hard disk - aliexpress
- Nmea4Wifi multiplexer https://vela-navega.com
- GPS/GLONASS U-blox v7 USB - amazon
- 2.4G Air Mouse with Touchpad Keyboard i8 Backlit Mini Wireless Keyboard - aliexpress
- Motorcycle mount 1 inch ball Aluminum Alloy 4 holes AMPS Hole, VESA75 - aliexpress
![IMG_20240612_124335](https://github.com/bareboat-necessities/lysmarine_gen/assets/34929784/0c3593b2-64cd-4b7e-a622-0880c410e50b)

**Reply by mgrouch** [reply]
There is 2024-06-10 version already. Just FYI

7” is a bit small so you might want to set font scaling to 0.85 in budgie Desktop settings if you see some forms not fitting into the screen.

**Reply by atchik6056** [reply]
what are the key changes between 2024-06-10 and 2024-04-20-r1252?

**Reply by mgrouch** [reply]
The are in release notes:

https://github.com/bareboat-necessities/lysmarine_gen/releases/tag/v2024-06-10

**Reply by atchik6056** [reply]
ok, thank you
I managed to upgrade signal k 2.7.2 to 2.8.2 in  bbn-bullseye-20230723 before from cli, preserving all settings, so I should be fine here
OS and firmware is already upgraded with apt and aptitude from defined repos.
PS:
going back to our discussion in bbn upgrade thread :-) sorry but I am not going to re-image, no point. but thank you though for letting me know.
I already lost SASplanet and other wine apps with reinstall to bookworm, your procedure with reinstalling wine for airmail and SASplanet does not work due to failing dependencies (huh it is so bloody typical, will discuss it in separate thread), and other some glitches in desktop tray, like nextcloud desktop icon is disappeared and etc
though there are some pluses, major: 2 fingers zoom started working in opencpn yeahhh! apparently there was something due to  DSI waveshare screen (which was discussed in the separate thread last year) in the previous install. Waveshare does not provide uninstall procedure xy*)(*.... Now I am using HDMI attached WT070B24 touchscreen and new kernel.
PS2.:
This setup is my guinea-pig project for my smaller race boat. I use it to test the upgrade and staff for my previous bbn-bullseye-20230723 and will repeat again 
Perhaps I ll publish other setup later, it is more interesting: bigger boat, more stuff is connected including 4G LTE network, AIS and etc

**Reply by mgrouch** [reply]
I’ll take a look at wine scripts. Did you use bookworm ones?

**Reply by mgrouch** [reply]
DSI screens issues were discussed here:

https://github.com/bareboat-necessities/lysmarine_gen/issues/416

There were some successes

**Reply by mgrouch** [reply]
nextcloud is not a part of BBN OS. 
not sure what’s the issue with its desktop icon.

**Reply by atchik6056** [reply]
> I’ll take a look at wine scripts. Did you use bookworm ones?

yes, https://bareboat-necessities.github.io/my-bareboat/airmail-install-bookworm.html
quickly, it boils down to :

```user@lysmarine:~ $ sudo apt-get -y install libegl-mesa0:armhf libgdm1:armhf libgl1-mesa-dri:armhf \
  libglapi-mesa:armhf libgles2-mesa:armhf libglu1-mesa:armhf libglx-mesa0:armhf \
  mesa-va-drivers:armhf mesa-vdpau-drivers:armhf mesa-vulkan-drivers:armhf \
  libsdl1.2debian:armhf libsdl2-2.0-0:armhf libudev1:armhf libllvm11:armhf \
  linux-libc-dev:armhf libncurses6:armhf libc6:armhf  libx11-6:armhf libgdk-pixbuf2.0-0:armhf \
  libgtk2.0-0:armhf libstdc++6:armhf libsdl2-2.0-0:armhf mesa-va-drivers:armhf \
  libsdl1.2-dev:armhf libsdl-mixer1.2:armhf libpng16-16:armhf libcal3d12v5:armhf \
  libsdl2-net-2.0-0:armhf libopenal1:armhf libsdl2-image-2.0-0:armhf \
  libvorbis-dev:armhf libcurl4:armhf libjpeg62:armhf  libudev1:armhf \
  libgl1-mesa-dev:armhf  libx11-dev:armhf libsmpeg0:armhf libavcodec58:armhf \
  libavformat58:armhf libswscale5:armhf libsdl2-image-2.0-0:armhf \
  libsdl2-mixer-2.0-0:armhf gcc-arm-linux-gnueabihf
[sudo] password for user: 
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libjpeg62-turbo:armhf' instead of 'libjpeg62:armhf'
E: Unable to locate package libllvm11:armhf
```
there is no package libllvm11:armhf  in bookworm, but libllvm14:armhf , however:
```user@lysmarine:~ $ sudo apt-get -y install libegl-mesa0:armhf libgdm1:armhf libgl1-mesa-dri:armhf   libglapi-mesa:armhf libgles2-mesa:armhf libglu1-mesa:armhf libglx-mesa0:armhf   mesa-va-drivers:armhf mesa-vdpau-drivers:armhf mesa-vulkan-drivers:armhf   libsdl1.2debian:armhf libsdl2-2.0-0:armhf libudev1:armhf libllvm14:armhf   linux-libc-dev:armhf libncurses6:armhf libc6:armhf  libx11-6:armhf libgdk-pixbuf2.0-0:armhf   libgtk2.0-0:armhf libstdc++6:armhf libsdl2-2.0-0:armhf mesa-va-drivers:armhf   libsdl1.2-dev:armhf libsdl-mixer1.2:armhf libpng16-16:armhf libcal3d12v5:armhf   libsdl2-net-2.0-0:armhf libopenal1:armhf libsdl2-image-2.0-0:armhf   libvorbis-dev:armhf libcurl4:armhf libjpeg62:armhf  libudev1:armhf   libgl1-mesa-dev:armhf  libx11-dev:armhf libsmpeg0:armhf libavcodec58:armhf   libavformat58:armhf libswscale5:armhf libsdl2-image-2.0-0:armhf   libsdl2-mixer-2.0-0:armhf gcc-arm-linux-gnueabihf
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'libjpeg62-turbo:armhf' instead of 'libjpeg62:armhf'
libgl1-mesa-dri:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
libgl1-mesa-dri:armhf set to manually installed.
libglapi-mesa:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
libglapi-mesa:armhf set to manually installed.
libglx-mesa0:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
libglx-mesa0:armhf set to manually installed.
mesa-va-drivers:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
mesa-va-drivers:armhf set to manually installed.
mesa-vdpau-drivers:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
mesa-vdpau-drivers:armhf set to manually installed.
mesa-vulkan-drivers:armhf is already the newest version (23.2.1-1~bpo12+rpt3).
mesa-vulkan-drivers:armhf set to manually installed.
libudev1:armhf is already the newest version (252.22-1~deb12u1).
libncurses6:armhf is already the newest version (6.4-4).
libc6:armhf is already the newest version (2.36-9+rpt2+deb12u7).
libx11-6:armhf is already the newest version (2:1.8.4-2+deb12u2).
libstdc++6:armhf is already the newest version (12.2.0-14).
libstdc++6:armhf set to manually installed.
libpng16-16:armhf is already the newest version (1.6.39-2).
libpng16-16:armhf set to manually installed.
libopenal1:armhf is already the newest version (1:1.19.1-2).
libcurl4:armhf is already the newest version (7.88.1-10+deb12u5).
libcurl4:armhf set to manually installed.
libjpeg62-turbo:armhf is already the newest version (1:2.1.5-2).
libjpeg62-turbo:armhf set to manually installed.
gcc-arm-linux-gnueabihf is already the newest version (4:12.2.0-3).
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
 libavcodec58:armhf : Depends: libaom0:armhf (>= 1.0.0) but it is not installable
                      Depends: libcodec2-0.9:armhf (>= 0.9.2) but it is not installable
                      Depends: libdav1d4:armhf (>= 0.2.1) but it is not installable
                      Depends: libvpx6:armhf (>= 1.6.0) but it is not installable
                      Depends: libwebp6:armhf (>= 0.5.1) but it is not installable
                      Depends: libx264-160:armhf (>= 2:0.160.3011+gitcde9a93) but it is not installable
                      Depends: libx265-192:armhf (>= 3.2) but it is not installable
 libavformat58:armhf : Depends: libsrt1.4-gnutls:armhf (>= 1.4.2) but it is not installable
E: Unable to correct problems, you have held broken packages.
user@lysmarine:~ $ 
```
I will dig into
if you want we can discuss in separate thread

**Comment by atchik6056** [comment]
> DSI screens issues were discussed here:
https://github.com/bareboat-necessities/lysmarine_gen/issues/416
There were some successes

I knew all this https://github.com/bareboat-necessities/lysmarine_gen/discussions/327
I had to comment out /hoem/user/.config/openbox/autostart:
```(openbox --replace) # &
#(evdev-rce || true) &
#(twofing --click=first --grab-by-disabling /dev/twofingtouch || true) &
```
otherwise touchscreen became crazy

**Reply by atchik6056** [reply]
>nextcloud is not a part of BBN OS.
not sure what’s the issue with its desktop icon.

I know, just listing, I install a lot of additional apps to be connected into my home base environment: cloud clients, vpn, monitoring, and so on and on
I am quite new to raspberry stuff though I have almost 40 y of all kind of linux/unix experience - my primary work.
An year ago I started comparing your BBN project and openplotter. I liked the idea of openplotter taking standard OS and converting it into openplotter apps. I have it on my ubuntu as backup. Though after some time I realised that their approach is not good for RPI. Their setup GUI is good looking windows for windows guys but sucks: I did configure network via cli, it worked, but as soon as I touched settings GUI, it overwrote everything without even informing me - windows habit.
I liked your approach more which is basically (as I understand): take standard OS and install all apps in standard ways as possible using install scripts from standard repos. This includes standard desktop - budgie. However this approach final result is appliance. And this s not your fault but RPI + linux nature. Though I disagree with you on the upgrade process. though I understand you do not have too many options here. However, for example, AIX upgrade mode "preservation install" is basically complete OS reinstall but it does not touch /home as a separate lv. Commercial SLES and RHEL are good going between releases too today, not like 10 years ago..
Another approach would be ansible yaml statefull deployment instead of scripts stateless run.
3rd one would be docker containers approach
just two cents though I do not know all the details of your install procedure.

**Reply by atchik6056** [reply]
nexcloud-destop tray icon is fixed
budgie problem https://discourse.ubuntubudgie.org/t/unable-to-enable-nautilus-nextcloud-on-ubuntu-budgie-desktop/5025/3

```
sudo apt install nemo-nextcloud
reboot
```

**Comment by mgrouch** [comment]
Here is one more setup

https://clubsearay.com/index.php?threads/raspberry-pi-on-the-boat.117159/#post-1485887

**Comment by mgrouch** [comment]
Here is one more interesting video 

https://youtu.be/BOPwfBOE1DE

**Comment by atchik6056** [comment]
final install of the above system 
![IMG_20240814_120957](https://github.com/user-attachments/assets/2b3b285e-a8f7-4f55-84ac-3c01a1cee5b1)

in addition I have managed to connect WT901C-232 and CP2102 USB \ TTL \ RS232 \ RS485 adapter, and got heeling data but not magnetic compass,as i was hoping, though there were some problems and workarounds: https://github.com/techgardeners/signalk-hwt901b-imu-plus/issues/1 to make it working (temporarily removed on the photo)

**Comment by kinyo666** [comment]
Here is my basic setup on the boat :

- Raspberry Pi 4B 4Gb (second hand, 68€)
- [uPerfect](https://www.uperfectmonitor.com/products/raspberry-pi-case-with-screen) 10" Touchscreen with integrated RPi case ([Cdiscount](https://www.cdiscount.com/informatique/ecrans-informatiques/moniteur-portable-tactile-10-1-pouces-1920x1200-h/f-10732-upe1690448266043.html#mpos=0%7Cmp), 140€)
- Bulk converter 5-32V-->12V@3A ([Cdiscount](https://www.cdiscount.com/bricolage/electricite/yosoo-module-convertisseur-cc-module-convertisseur/f-16614-yos1702845225451.html#mpos=0%7Cmp), 16€)
- 1 Wago connector ([Cdiscount](https://www.cdiscount.com/bricolage/electricite/mini-borne-de-connexion-a-levier-3-x-0-2-4-mm2-p/f-1661412-per5410329642990.html), 1,60€)
- 1 fuse 3A ([Norauto](https://www.norauto.fr/p/porte-fusible-15-a-221869.html), 3,20€)
- Power cable with DC plug 5.5/2.5 and case from an old laptop

![20240824_171227](https://github.com/user-attachments/assets/2cde2849-dd60-4511-b454-57882aaa742a)

![20240824_185032](https://github.com/user-attachments/assets/d4a5fa38-5841-4358-acfc-cf8a07ebc876)

I have also customized the BBN desktop with new titles (in french :) ) and icons :
![bbnlauncher_new_menu](https://github.com/user-attachments/assets/a0acbdad-6762-4942-88e2-0ef78d36ca8d)

![bbnlauncher_new_menu2](https://github.com/user-attachments/assets/11b6cf41-94e1-4f05-a738-624bdae1bcfc)


Next steps :
- Add an ESP32 with SensESP SDK to digitalize the engine instrument panel (voltmeter, water temperature, oil pressure, rev counter)
- Add INA219 (or INA3221) sensors to monitor fuel tank, grey and fresh water level
- Add DS18B20 temperature sensor to monitor engine room
- Check if my old Raymarine ST6000 autopilot controller can be used remotely

**Reply by mgrouch** [reply]
I’ve also made heave sensor (measures boat movement in waves)

It only needs m5atom-S3 from m5stack. 

https://github.com/bareboat-necessities/bbn-wave-period-esp32/tree/v0.1.0

Cheers!

**Reply by kinyo666** [reply]
Yes, i'm following your work on facebook ;)
Have you already experimented it on the sea ?

**Reply by mgrouch** [reply]
> Yes, i'm following your work on facebook ;) Have you already experimented it on the sea ?

I’ve just made it work in decent shape today. 
I haven’t experimented in sea trials, but I’d love to hear others experiences. I tested it on a broomstick at home imitating smaller waves.

**Comment by mgrouch** [comment]
Here is another video of another installation 

https://youtu.be/zE4mcTktGBY?feature=shared

and this thread:

https://www.thehulltruth.com/marine-electronics-forum/1320681-smart-boat-projects.html
